### PR TITLE
IBFT private state commit issue in a single validator node network

### PR DESCRIPTION
**Pull request by vsmk98**
The private state does not get committed for validator node which results in a node crash upon restart
Steps to simulate

Bring up a Quorum IBFT network with only one node as validator
From the validator node perform a private transaction
Restart the network
Upon restart the validator node crashes with the below message

```WARN [01-14|11:11:11.192] Rewinding blockchain                     target=0
panic: interface conversion: interface {} is nil, not *types.Block
goroutine 1 [running]:
github.com/ethereum/go-ethereum/core.(*BlockChain).CurrentBlock(...)
        /Users/saiv/go/src/github.com/ethereum/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/core/blockchain.go:368
github.com/ethereum/go-ethereum/core.(*BlockChain).SetHead(0xc00019a000, 0x0, 0x0, 0x0)
        /Users/saiv/go/src/github.com/ethereum/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/core/blockchain.go:304 +0xabc
github.com/ethereum/go-ethereum/core.(*BlockChain).ResetWithGenesisBlock(0xc00019a000, 0xc0001c2240, 0x0, 0x0)
        /Users/saiv/go/src/github.com/ethereum/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/core/blockchain.go:433 +0x54
github.com/ethereum/go-ethereum/core.(*BlockChain).Reset(0xc00019a000, 0x2b, 0xc0004e6038)
        /Users/saiv/go/src/github.com/ethereum/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/core/blockchain.go:426 +0x37
github.com/ethereum/go-ethereum/core.(*BlockChain).loadLastState(0xc00019a000, 0x0, 0xc0001c2240)
        /Users/saiv/go/src/github.com/ethereum/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/core/blockchain.go:241 +0x43e
github.com/ethereum/go-ethereum/core.NewBlockChain(0x50af020, 0xc0002e6fd0, 0xc0000349c0, 0xc0002e7080, 0x7e21ec0, 0xc0000e8fc0, 0x0, 0x0, 0x0, 0x0, ...)
        /Users/saiv/go/src/github.com/ethereum/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/core/blockchain.go:187 +0x5ef
github.com/ethereum/go-ethereum/eth.New(0xc000229c80, 0xc000108000, 0x400e488, 0x30, 0x4da1f20)
        /Users/saiv/go/src/github.com/ethereum/go-ethereum/build/_workspace/src/github.```

### CODE REVIEW

File: core/blockchain.go [hunk 1]
```diff
@@ -927,6 +927,23 @@ func (bc *BlockChain) WriteBlockWithState(block *types.Block, receipts []*types.
 	bc.wg.Add(1)
 	defer bc.wg.Done()
 
+	// Quorum
```
Comment (jpmsam) [inline core/blockchain.go:939]: We should persist to DB after we acquire the lock `bc.mu.Lock()`

Comment (vsmk98) [reply core/blockchain.go:939]: changed to persist db after lock has been acquired

**Review by trung — APPROVED** [review]
LGTM

**Review by jbhurat — APPROVED** [review]
LGTM

**Review by jpmsam — APPROVED** [review]

**Pull request merged**